src/library/searchquery.cpp

```diff
@@ -166,6 +173,38 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument,
+                                 const CrateStorage* pCrateStorage)
+    : m_database(database),
+      m_argument(argument),
+      m_pCrateStorage(pCrateStorage) {
+    init();
+}
+
+void CrateFilterNode::init() {
+    CrateTrackSelectResult crateTracks(
+        m_pCrateStorage->selectTracksSortedByCrateNameLike(m_argument));
+
+    while (crateTracks.next()) {
+        m_trackIds.push_back(crateTracks.trackId());
+    }
+}
+
+bool CrateFilterNode::match(const TrackPointer& pTrack) const {
+    return std::binary_search(m_trackIds.begin(), m_trackIds.end(), pTrack->getId());
+}
+
+QString CrateFilterNode::toSql() const {
```

Comment:
Wouldn't this run the subselect query twice, once as a subquery in the main search filter query, and again to get the list of TrackIds for match()? That would be faster than building a QStringList from the vector of TrackIds?